tlf resets exchange message to zero (CQWW CW)

Checking the current git version for CQWW CW. It seems that the zone info is initially set correctly but after pressing return, it resets to zero. Tlf version that comes with debian keeps the zone there correctly.

make check bombs with:
  CCLD     run_searchlog
/usr/bin/ld: run_searchlog.o:(.data+0x328): undefined reference to `test_ZoneFromCountry'
/usr/bin/ld: run_searchlog.o:(.data+0x350): undefined reference to `test_ZoneFromExchange'
/usr/bin/ld: run_searchlog.o:(.data+0x378): undefined reference to `test_ZoneFromLog_mixedmode'
/usr/bin/ld: run_searchlog.o:(.data+0x3a0): undefined reference to `test_ZoneFromLog'
/usr/bin/ld: ../src/searchlog.o: in function `searchlog':
/home/eloranta/c/tlf/src/searchlog.c:687: undefined reference to `get_proposed_exchange'
collect2: error: ld returned 1 exit status
make[2]: *** [Makefile:915: run_searchlog] Error 1
make[2]: Leaving directory '/home/eloranta/c/tlf/test'
make[1]: *** [Makefile:1437: check-am] Error 2
make[1]: Leaving directory '/home/eloranta/c/tlf/test'
make: *** [Makefile:475: check-recursive] Error 1

Here is the logfile:
[==========] Running 4 test(s).
[ RUN      ] test_getexchange_arrlss
[       OK ] test_getexchange_arrlss
[ RUN      ] test_getexchange_cqww
[  ERROR   ] --- "00" != "12"
[   LINE   ] --- test_getexchange.c:261: error: Failure!
[  FAILED  ] test_getexchange_cqww
[ RUN      ] test_getexchange_serial_section
[       OK ] test_getexchange_serial_section
[ RUN      ] test_getexchange_sectn_mult
[       OK ] test_getexchange_sectn_mult
[==========] 4 test(s) run.
[  PASSED  ] 3 test(s).
[  FAILED  ] 1 test(s), listed below:
[  FAILED  ] test_getexchange_cqww

 1 FAILED TEST(S)
FAIL run_getexchange (exit status: 1)

--- a/src/getexchange.c
+++ b/src/getexchange.c
@@ -503,7 +503,7 @@ static void checkexchange_cqww(char *comment, bool interactive) {
 	// get zone nr, use fix if available
 	index = g_match_info_fetch(match_info, 1);
 	gchar *index_fix = g_match_info_fetch(match_info, 3);
-	if (index_fix != NULL) {
+	if (index_fix != NULL && strlen(index_fix) >= 1 && strlen(index_fix) <= 4) {
 	    g_free(index);
 	    index = index_fix;
 	}

